yarn-project/archiver/src/archiver/eth_log_handlers.ts

```diff
@@ -43,16 +43,23 @@ export function processPendingL1ToL2MessageAddedLogs(
 }
 /**
  * Processes newly received UnverifiedData logs.
+ * @param blockHashMapping - A mapping from block number to relevant block hash.
  * @param logs - ContractDeployment logs.
  * @returns The set of retrieved contract public data items.
  */
 export function processContractDeploymentLogs(
+  blockHashMapping: { [key: number]: Buffer | undefined },
   logs: Log<bigint, number, undefined, typeof UnverifiedDataEmitterAbi, 'ContractDeployment'>[],
 ): [ContractPublicData[], number][] {
   const contractPublicData: [ContractPublicData[], number][] = [];
   for (let i = 0; i < logs.length; i++) {
     const log = logs[i];
     const l2BlockNum = Number(log.args.l2BlockNum);
+    const blockHash = Buffer.from(hexToBytes(log.args.l2BlockHash));
+    const expectedBlockHash = blockHashMapping[Number(l2BlockNum)];
```

Comment:
nit:
```suggestion
    const expectedBlockHash = blockHashMapping[l2BlockNum];
```
redundant case, its already casted above